common/namespace/registry.go

```diff
@@ -489,3 +529,23 @@ func (r *registry) triggerNamespaceChangeCallbackLocked(
 		callback(prevNamespaces, nextNamespaces)
 	}
 }
+
+type fetchBy func(*persistence.GetNamespaceRequest)
+
+func byName(name string) func(*persistence.GetNamespaceRequest) {
+	return func(req *persistence.GetNamespaceRequest) { req.Name = name }
+}
+
+func byID(id string) func(*persistence.GetNamespaceRequest) {
+	return func(req *persistence.GetNamespaceRequest) { req.ID = id }
+}
```

Comment:
It seems simpler for these to return a new `*persistence.GetNamespaceRequest`. Is there a case where you need to use both of them together?